### PR TITLE
feat(unit-price): Add unit amount cents to subscription fee

### PR DESCRIPTION
**Pull request by rsempe**
## Roadmap Task

👉  [https://getlago.canny.io/feature-requests/p/{{FEATURE_SLUG}}](https://getlago.canny.io/feature-requests/p/display-unit-price-fee-breakdown)

## Context

Invoices are not compliant as we’re not displaying unit prices of item on them.

The goal is to be compliant by adding unit price on invoice payload and pdf.
- Subscription invoices
- Charge (pay in advance) invoices

## Description

The goal of this PR is to:
- convert unit amount cents to decimal
- compute unit amount cents to subscription's fee

### CODE REVIEW

File: db/migrate/20231020091031_convert_unit_amount_cents_to_decimal.rb [hunk 1]
```diff
@@ -0,0 +1,7 @@
+# frozen_string_literal: true
+
+class ConvertUnitAmountCentsToDecimal < ActiveRecord::Migration[7.0]
+  def change
+    change_column :fees, :unit_amount_cents, :decimal, precision: 30, scale: 5, null: false
```
Comment (jdenquin) [inline db/migrate/20231020091031_convert_unit_amount_cents_to_decimal.rb:5]: Rails/ReversibleMigration: change_column is not reversible. (https://rails.rubystyle.guide#reversible-migration, https://api.rubyonrails.org/classes/ActiveRecord/Migration/CommandRecorder.html)

File: db/migrate/20231020091031_convert_unit_amount_cents_to_decimal.rb [hunk 2]
```diff
@@ -0,0 +1,7 @@
...
+  end
+end
```
Comment (jdenquin) [inline db/migrate/20231020091031_convert_unit_amount_cents_to_decimal.rb:7]: Layout/TrailingEmptyLines: Final newline missing. (https://rubystyle.guide#newline-eof)

File: spec/services/fees/subscription_service_spec.rb [hunk 3]
```diff
@@ -52,17 +52,17 @@
...
-      end
+      expect(result.fee).to have_attributes(
+        id: String,
+        invoice_id: invoice.id,
+        amount_cents: plan.amount_cents,
+        amount_currency: plan.amount_currency,
+        units: 1,
+        events_count: nil,
+        payment_status: 'pending',
+        unit_amount_cents: plan.amount_cents
```
Comment (jdenquin) [inline spec/services/fees/subscription_service_spec.rb:64]: Style/TrailingCommaInArguments: Put a comma after the last parameter of a multiline method call. (https://rubystyle.guide#no-trailing-params-comma)

File: spec/services/fees/subscription_service_spec.rb [hunk 4]
```diff
@@ -76,12 +76,12 @@
...
-          created_fee = result.fee
 
-          aggregate_failures do
-            expect(created_fee.id).not_to be_nil
-            expect(created_fee.amount_cents).to eq(90)
-          end
+          expect(result.fee).to have_attributes(
+            id: String,
+            amount_cents: 90,
+            unit_amount_cents: 90
```
Comment (jdenquin) [inline spec/services/fees/subscription_service_spec.rb:83]: Style/TrailingCommaInArguments: Put a comma after the last parameter of a multiline method call. (https://rubystyle.guide#no-trailing-params-comma)

File: spec/services/fees/subscription_service_spec.rb [hunk 5]
```diff
@@ -124,15 +124,15 @@
...
-            expect(created_fee.units).to eq(1)
-          end
+
+          expect(result.fee).to have_attributes(
+            id: String,
+            invoice_id: invoice.id,
+            amount_cents: plan.amount_cents,
+            amount_currency: plan.amount_currency,
+            unit_amount_cents: plan.amount_cents,
+            units: 1
```
Comment (jdenquin) [inline spec/services/fees/subscription_service_spec.rb:134]: Style/TrailingCommaInArguments: Put a comma after the last parameter of a multiline method call. (https://rubystyle.guide#no-trailing-params-comma)

File: spec/services/fees/subscription_service_spec.rb [hunk 6]
```diff
@@ -204,15 +204,14 @@
...
-            expect(created_fee.amount_currency).to eq(plan.amount_currency)
-            expect(created_fee.units).to eq(1)
-          end
+
+          expect(result.fee).to have_attributes(
+            id: String,
+            invoice_id: invoice.id,
+            amount_cents: 71,
+            amount_currency: plan.amount_currency,
+            units: 1
```
Comment (jdenquin) [inline spec/services/fees/subscription_service_spec.rb:213]: Style/TrailingCommaInArguments: Put a comma after the last parameter of a multiline method call. (https://rubystyle.guide#no-trailing-params-comma)

File: spec/services/fees/subscription_service_spec.rb [hunk 7]
```diff
@@ -318,15 +317,15 @@
...
-            expect(created_fee.units).to eq(1)
-          end
+
+          expect(result.fee).to have_attributes(
+            id: String,
+            invoice_id: invoice.id,
+            amount_cents: plan.amount_cents,
+            amount_currency: plan.amount_currency,
+            unit_amount_cents: plan.amount_cents,
+            units: 1
```
Comment (jdenquin) [inline spec/services/fees/subscription_service_spec.rb:327]: Style/TrailingCommaInArguments: Put a comma after the last parameter of a multiline method call. (https://rubystyle.guide#no-trailing-params-comma)

File: spec/services/fees/subscription_service_spec.rb [hunk 8]
```diff
@@ -405,15 +404,14 @@
...
-            expect(created_fee.amount_currency).to eq(plan.amount_currency)
-            expect(created_fee.units).to eq(1)
-          end
+
+          expect(result.fee).to have_attributes(
+            id: String,
+            invoice_id: invoice.id,
+            amount_cents: 55,
+            amount_currency: plan.amount_currency,
+            units: 1
```
Comment (jdenquin) [inline spec/services/fees/subscription_service_spec.rb:413]: Style/TrailingCommaInArguments: Put a comma after the last parameter of a multiline method call. (https://rubystyle.guide#no-trailing-params-comma)

File: spec/services/fees/subscription_service_spec.rb [hunk 9]
```diff
@@ -549,14 +547,15 @@
...
-            expect(created_fee.amount_cents).to eq(plan.amount_cents)
-            expect(created_fee.amount_currency).to eq(plan.amount_currency)
-          end
+          expect(result.fee).to have_attributes(
+            id: String,
+            invoice_id: invoice.id,
+            amount_cents: plan.amount_cents,
+            amount_currency: plan.amount_currency,
+            unit_amount_cents: plan.amount_cents,
+            units: 1
```
Comment (jdenquin) [inline spec/services/fees/subscription_service_spec.rb:557]: Style/TrailingCommaInArguments: Put a comma after the last parameter of a multiline method call. (https://rubystyle.guide#no-trailing-params-comma)

File: spec/services/fees/subscription_service_spec.rb [hunk 10]
```diff
@@ -586,14 +585,14 @@
...
-            expect(created_fee.invoice_id).to eq(invoice.id)
-            expect(created_fee.amount_cents).to eq(80)
-            expect(created_fee.amount_currency).to eq(plan.amount_currency)
-          end
+          expect(result.fee).to have_attributes(
+            id: String,
+            invoice_id: invoice.id,
+            amount_cents: 80,
+            amount_currency: plan.amount_currency,
+            units: 1
```
Comment (jdenquin) [inline spec/services/fees/subscription_service_spec.rb:594]: Style/TrailingCommaInArguments: Put a comma after the last parameter of a multiline method call. (https://rubystyle.guide#no-trailing-params-comma)

File: spec/services/fees/subscription_service_spec.rb [hunk 11]
```diff
@@ -791,14 +790,14 @@
...
-        expect(created_fee.invoice_id).to eq(invoice.id)
-        expect(created_fee.amount_cents).to eq(65)
-        expect(created_fee.amount_currency).to eq(plan.amount_currency)
-      end
+      expect(result.fee).to have_attributes(
+        id: String,
+        invoice_id: invoice.id,
+        amount_cents: 65,
+        amount_currency: plan.amount_currency,
+        units: 1
```
Comment (jdenquin) [inline spec/services/fees/subscription_service_spec.rb:799]: Style/TrailingCommaInArguments: Put a comma after the last parameter of a multiline method call. (https://rubystyle.guide#no-trailing-params-comma)

File: spec/services/fees/subscription_service_spec.rb [hunk 12]
```diff
@@ -819,14 +818,14 @@
...
-          expect(created_fee.invoice_id).to eq(invoice.id)
-          expect(created_fee.amount_cents).to eq(65)
-          expect(created_fee.amount_currency).to eq(plan.amount_currency)
-        end
+        expect(result.fee).to have_attributes(
+          id: String,
+          invoice_id: invoice.id,
+          amount_cents: 65,
+          amount_currency: plan.amount_currency,
+          units: 1
```
Comment (jdenquin) [inline spec/services/fees/subscription_service_spec.rb:827]: Style/TrailingCommaInArguments: Put a comma after the last parameter of a multiline method call. (https://rubystyle.guide#no-trailing-params-comma)

File: spec/services/fees/subscription_service_spec.rb [hunk 13]
```diff
@@ -845,14 +844,14 @@
...
-          expect(created_fee.invoice_id).to eq(invoice.id)
-          expect(created_fee.amount_cents).to eq(43)
-          expect(created_fee.amount_currency).to eq(plan.amount_currency)
-        end
+        expect(result.fee).to have_attributes(
+          id: String,
+          invoice_id: invoice.id,
+          amount_cents: 43,
+          amount_currency: plan.amount_currency,
+          units: 1
```
Comment (jdenquin) [inline spec/services/fees/subscription_service_spec.rb:853]: Style/TrailingCommaInArguments: Put a comma after the last parameter of a multiline method call. (https://rubystyle.guide#no-trailing-params-comma)

File: spec/services/fees/subscription_service_spec.rb [hunk 14]
```diff
@@ -863,14 +862,14 @@
...
         result = fees_subscription_service.create
-        created_fee = result.fee
-
-        aggregate_failures do
-          expect(created_fee.id).not_to be_nil
-          expect(created_fee.invoice_id).to eq(invoice.id)
-          expect(created_fee.amount_cents).to eq(65)
-          expect(created_fee.amount_currency).to eq(plan.amount_currency)
-        end
+
```
Comment (jdenquin) [inline spec/services/fees/subscription_service_spec.rb:865]: Layout/TrailingWhitespace: Trailing whitespace detected. (https://rubystyle.guide#no-trailing-whitespace)

File: spec/services/fees/subscription_service_spec.rb [hunk 15]
```diff
@@ -863,14 +862,14 @@
...
-          expect(created_fee.amount_cents).to eq(65)
-          expect(created_fee.amount_currency).to eq(plan.amount_currency)
-        end
+       
+        expect(result.fee).to have_attributes(
+          id: String,
+          invoice_id: invoice.id,
+          amount_cents: 65,
+          amount_currency: plan.amount_currency,
+          units: 1
```
Comment (jdenquin) [inline spec/services/fees/subscription_service_spec.rb:871]: Style/TrailingCommaInArguments: Put a comma after the last parameter of a multiline method call. (https://rubystyle.guide#no-trailing-params-comma)

File: spec/services/fees/subscription_service_spec.rb [hunk 16]
```diff
@@ -940,14 +939,14 @@
...
-        expect(created_fee.invoice_id).to eq(invoice.id)
-        expect(created_fee.amount_cents).to eq(55)
-        expect(created_fee.amount_currency).to eq(plan.amount_currency)
-      end
+      expect(result.fee).to have_attributes(
+        id: String,
+        invoice_id: invoice.id,
+        amount_cents: 55,
+        amount_currency: plan.amount_currency,
+        units: 1
```
Comment (jdenquin) [inline spec/services/fees/subscription_service_spec.rb:948]: Style/TrailingCommaInArguments: Put a comma after the last parameter of a multiline method call. (https://rubystyle.guide#no-trailing-params-comma)

File: spec/services/fees/subscription_service_spec.rb [hunk 17]
```diff
@@ -968,14 +967,14 @@
...
-          expect(created_fee.invoice_id).to eq(invoice.id)
-          expect(created_fee.amount_cents).to eq(55)
-          expect(created_fee.amount_currency).to eq(plan.amount_currency)
-        end
+        expect(result.fee).to have_attributes(
+          id: String,
+          invoice_id: invoice.id,
+          amount_cents: 55,
+          amount_currency: plan.amount_currency,
+          units: 1
```
Comment (jdenquin) [inline spec/services/fees/subscription_service_spec.rb:976]: Style/TrailingCommaInArguments: Put a comma after the last parameter of a multiline method call. (https://rubystyle.guide#no-trailing-params-comma)

**Review by vincent-pochet — APPROVED** [review]

**Pull request merged**